ggml/src/ggml-vulkan.cpp

```diff
@@ -2042,9 +2042,9 @@ void ggml_vk_instance_init() {
 }
 
 static void ggml_vk_init(ggml_backend_vk_context * ctx, size_t idx) {
-    GGML_ASSERT(idx < vk_instance.device_indices.size());
     VK_LOG_DEBUG("ggml_vk_init(" << ctx->name << ", " << idx << ")");
     ggml_vk_instance_init();
+    GGML_ASSERT(idx < vk_instance.device_indices.size());
```

Comment:
Is this change necessary? The original code works in https://github.com/ggerganov/llama.cpp, not sure why we would need to change it here

Reply:
I changed it so `ggml_vk_instance_init()` would populate `device_indices` right before the assertion, and it would avoid another vk-specific call in whisper.cpp. Without this, the assertion failed every time, no other code had attempted to initialize devices.

I didn't look at how llama.cpp did it, but I can check it out.

Reply:
I see, seems the change is good. cc @0cc4m to confirm that it's OK to move the assert after the init call

Reply:
Yeah, that's fine. Makes more sense that way, too.